server/models/types.ts

```diff
@@ -110,7 +110,7 @@ export type DetectorResultsQueryParams = {
 };
 
 export type MDSQueryParams = {
-  dataSourceId?: string;
+  dataSourceId: string;
```

Comment:
What if `dataSourceId` were falsy here and `Local cluster` is disabled?

Reply:
as I replied above, I think we have talked about this before and the decision is data source picker will throw error banner in that case